tracing/src/macros.rs

```diff
@@ -632,35 +630,38 @@ macro_rules! event {
         $crate::event!(target: $target, parent: $parent, $lvl, { $($arg)+ })
     );
     (target: $target:expr, $lvl:expr, { $($fields:tt)* } )=> ({
-        $crate::__tracing_log!(
+        use $crate::__macro_support::Callsite as _;
+        static CALLSITE: $crate::__macro_support::MacroCallsite = $crate::callsite2! {
+            name: concat!(
+                "event ",
+                file!(),
+                ":",
+                line!()
+            ),
+            kind: $crate::metadata::Kind::EVENT,
             target: $target,
-            $lvl,
-            $($fields)*
-        );
-        if $crate::level_enabled!($lvl) {
-            use $crate::__macro_support::Callsite as _;
-            static CALLSITE: $crate::__macro_support::MacroCallsite = $crate::callsite2! {
-                name: concat!(
-                    "event ",
-                    file!(),
-                    ":",
-                    line!()
-                ),
-                kind: $crate::metadata::Kind::EVENT,
+            level: $lvl,
+            fields: $($fields)*
+        };
+        (|value_set: $crate::field::ValueSet| {
```

Comment:
It appears to be passing, which slightly surprises me because I can't see what's changed, I'll dig into this more to work out why, just in case the test is missing something. (oooh, it might be because value_set is constructed outside the closure)
I need to add some comments about this but the reason for the immediately invoked closures is to extend the lifetime of the temporaries created by valueset.
This doesn't compile because the string's are dropped before we get to baz.
```rust
let x = &[&String::from("foo"), &String::from("bar")];
baz(x);
```
This however does compile:
```rust
(|x| {
    baz(x);
})(&[&String::from("foo"), &String::from("bar")]);
```